Tests/NIOHTTPClientTests/SwiftNIOHTTPTests.swift

```diff
@@ -170,6 +170,32 @@ class SwiftHTTPTests: XCTestCase {
         response = try httpClient.get(url: "http://localhost:\(httpBin.port)/redirect/https?port=\(httpsBin.port)").wait()
         XCTAssertEqual(response.status, .ok)
     }
+    
+    func testHttpHostRedirect() throws {
+        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew,
+                                    configuration: HTTPClient.Configuration(certificateVerification: .none, followRedirects: true))
+        
+        defer {
+            try! httpClient.syncShutdown()
+        }
+        
+        let response = try httpClient.get(url: "https://httpbin.org/redirect-to?url=https%3A%2F%2Fwww.httpbin.org%2Fheaders").wait()
+        guard let body = response.body else {
+            XCTFail("The target page should have a body containing request headers")
+            return
+        }
+        let data = body.withUnsafeReadableBytes {
```

Comment:
I'd recommend using `readData` from the `NIOFoundationCompat` module. That method uses Data's `bytesNoCopy` initializer. It's also good practice to use the `read*` methods on `ByteBuffer` in general. 

Not a huge deal here since this is a test though, just FYI. :)

```swift
import NIOFoundationCompat

guard var body = response.body else { ... }
guard let data = body.readData(length: body.readableBytes) else { ... }
```

Reply:
Thank you! Interesting 🙂 
Fixed with latest commit